test/e2e/apigatewayv2/tests/test_api.py

```diff
@@ -0,0 +1,539 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the API Gateway V2
+"""
+
+import logging
+import time
+
+import boto3
+import pytest
+import requests
+import random
+import string
+
+from apigatewayv2 import SERVICE_NAME, service_marker
+from apigatewayv2.bootstrap_resources import get_bootstrap_resources
+from apigatewayv2.replacement_values import REPLACEMENT_VALUES
+import apigatewayv2.tests.helper as helper
+from apigatewayv2.tests.helper import ApiGatewayValidator
+from common import k8s
+from common.aws import get_aws_region, get_aws_account_id
+from common.resources import load_resource_file
+
+DELETE_WAIT_AFTER_SECONDS = 10
+UPDATE_WAIT_AFTER_SECONDS = 10
+DEPLOYMENT_WAIT_AFTER_SECONDS = 10
+
+apigw_validator = ApiGatewayValidator(boto3.client('apigatewayv2'))
+test_resource_values = REPLACEMENT_VALUES.copy()
+
+
+@pytest.fixture(scope="module")
+def api_resource():
+    api_resource_name = test_resource_values['API_NAME']
+    api_ref, api_data = helper.api_ref_and_data(api_resource_name=api_resource_name,
+                                                replacement_values=test_resource_values)
+    if k8s.get_resource_exists(api_ref):
+        pytest.fail(f"expected {api_resource_name} to not exist. Did previous test cleanup?", False)
+    logging.debug(f"http api resource. name: {api_resource_name}, data: {api_data}")
+
+    k8s.create_custom_resource(api_ref, api_data)
+    cr = k8s.wait_resource_consumed_by_controller(api_ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(api_ref)
+
+    api_id = cr['status']['apiID']
+    test_resource_values['API_ID'] = api_id
```

Comment:
You're modifying the default replacement values? Might be worth putting `REPLACEMENT_VALUES.copy()` into this fixture, rather than globally. Just worried about the situation where multitasking the tests calls them out of order and you pass around modified values.

Reply:
Oh I think I see, you want to use the modified versions in all places? Why not just return a copy of the `REPLACEMENT_VALUES` as a return value from this fixture? Or just return the `api_id` as a single value?